tutorials-v5/brmesolve_phonon_interaction.md

```diff
@@ -0,0 +1,263 @@
+
+$$ H_\textrm{phonon}=\hbar J(\omega)\sigma^\dagger \sigma,$$
+
+where $J(\omega)$ is the spectral density of the coupling.
+
+```python
+# operator that couples the quantum dot to acoustic phonons
+a_op = sm.dag() * sm
+
+# This spectrum is a displaced gaussian multiplied by w^3, which
+# models coupling to LA phonons. The electron and hole
+# interactions contribute constructively.
+
+
+# fitting parameters ae/ah
+ah = 1.9e-9  # m
+ae = 3.5e-9  # m
+# GaAs material parameters
+De = 7
+Dh = -3.5
+v = 5110  # m/s
+rho_m = 5370  # kg/m^3
+# Other Constants
+hbar = 1.05457173e-34  # Js
+T = 4.2  # Kelvin, temperature
+
+# constants for temperature dependence
+t1 = 0.6582119
+t2 = 0.086173
+
+# General J factor
+J = (
+    "(1.6 * 1e-13 * w**3) / (4 * pi**2 * rho_m * hbar * v**5) * "
+    + "(De * exp(-(w * 1e12 * ae * 0.5 / v)**2) - "
+    + "Dh * exp(-(w * 1e12 * ah * 0.5 / v)**2))**2"
+)
+
+# Term for positive frequencies
+JT_p = (
+    J
+    + "* (1 + exp(-w*t1/(T*t2)) / \
+          (1-exp(-w*t1/(T*t2))))"
+)
+
+# Term for negative frequencies
+JT_m = (
+    "-1.0* "
+    + J
+    + "* exp(w*t1/(T*t2)) / \
+            (1-exp(w*t1/(T*t2)))"
+)
+
+
+# define spectra with variable names
+spectra_cb = "(w > 0) * " + JT_p + "+ (w < 0) * " + JT_m
+
+# add a check for min size of w to avoid numerical problems
+spectra_cb = "0 if (w > -1e-4 and w < 1e-4) else " + spectra_cb
+
+# define string with numerical values expect for w
+constants = ["ah", "ae", "De", "Dh", "v", "rho_m", "hbar", "T", "t1", "t2"]
+spectra_cb_numerical = spectra_cb
+for c in constants:
+    # replace constants with numerical value
+    spectra_cb_numerical = spectra_cb_numerical.replace(c, str(eval(c)))
+```
+
+## Visualize the dot-phonon interaction spectrum
+
+$J(\omega)$ has two components that give rise to its shape: a rising component due to the increasing acoustic phonon density of states and a roll-off that occurs due to the physical size of the quantum dot.
+
+```python
+# frequency list
+spec_list = np.linspace(-5, 10, 200)
+
+# define shortcuts to evaluate string
+pi = np.pi
+exp = np.exp
+
+# plot the spectrum J(w)
+plt.figure(figsize=(8, 5))
+plt.plot(spec_list, [eval(spectra_cb.replace("w", str(_))) for _ in spec_list])
+plt.xlim(-5, 10)
+plt.xlabel("$\\omega$ [THz]")
+plt.ylabel("$J(\\omega)$ [THz]")
+plt.title("Quantum-dot-phonon interaction spectrum");
+```
+
+## Calculate the pulse-system interaction dynamics
+
+The Bloch-Redfield master equation solver takes the Hamiltonian time-dependence in list-string format. We calculate the final population at the end of the interaction of the pulse with the system, which represents the population initialized into the excited state.
+
+```python
+# we will calculate the dot population expectation value
+e_ops = [sm.dag() * sm]
+
+# sample brmesolve call to initialize coefficients
+H = [wd_list[0] * H_S, [Om_list[0] * H_I, pulse_shape]]
```

Comment:
I don't think this does anything useful in either version 4 or 5 now. We should at least try removing i and seeing what breaks. Originally it was probably a workaround for how coefficients were created and cleaned up in version 4.

There is a chance that I am wrong though and that `parfor` without first triggering coefficient compilation will fail, but in that case it would be good to know.

Reply:
As you wrote, there is a problem with running `parfor` in combination with `brmesolve` due to the coefficient compilation. I inserted this sample call for the coefficient initialization.

Reply:
Could we then just explain the issue here more clearly in the comment? I don't think it would be clear to someone who hadn't experienced this issue why the initialization is necessary. I also wonder if there is a more efficient way to initialize the coefficient? E.g. in QuTiP v5, one could just do `H = QobjEvo(...)` and that should initialize the coefficient, I think?

Reply:
This is a QuTiP v5 problem only. I tried running `H = QobjEvo(...)` before using `parfor`, but it also resulted in `brmesolve` not finding some `.pyx` file. Maybe this is due to the changing coefficients in the Hamiltonian?

I'll explain the sample call of `brmesolve` in the comment before.